### ISSUE TITLE
document or merge big/json into big

### ISSUE THREAD

**Issue by bo-tato** [crystal-lang/crystal#13746]
I was dealing with json with arbitrarily large numbers, and wrote a custom `JSON::PullParser` for the Big* types. I was going to submit a feature request with it, when I search github issues and realize this has already been implemented, if you `require "big/json"`. But this isn't documented anywhere I can find. As it's only a few lines of code I think the best solution would be to just have it included when you `require "big"`. Otherwise we should add documentation to the Big* pages saying if you want to deserialize json you need to also require "big/json"

**Comment by Blacksmoke16** [comment]
Documenting the file is probably the better option. Otherwise requiring `big` would _always_ require `json` even if you're not using it. _Maybe_ tree shaking would clean it up inherently, but would rather be a bit more explicit when it comes to that kind of stuff anyway.

Also should document these kind of extension files within the `JSON` docs as well.

**Comment by straight-shoota** [comment]
Documenting is certainly the *first* option.

However, I would also consider alternative options to improve the developer experience and make these requires more implicit.
It feels weird that I have to `big/json` explicitly just to get the serialization behaviour of `Big`.
It shouldn't add much overhead to parse the JSON library. As long as it's not used, the semantic effort should be negligible.

**Comment by straight-shoota** [comment]
Yeah, I would not have `require "yaml"` pull in all the stuff. Instead, `require "big"` would include `require "big/yaml"`.

**Comment by HertzDevil** [comment]
That is because `Big` needs global configuration (of libgmp's allocation functions) and `YAML` doesn't; you cannot use the big numbers without the program calling `LibGMP.set_memory_functions`.

Now let's say XML serialization is supported in the standard library, and UUID supports serialization. XML calls `LibXML.xmlGcMemSetup`, UUID requires no such configuration. Would that mean `require "xml"` includes `require "uuid/xml"`?
